contents/blog/welcome-to-deskhog.mdx

```diff
@@ -0,0 +1,80 @@
+
+Assuming you still like us when you pick your next project.
+
+Because this is true, I’ve been able to convince whole teams and all our executives to support me in doing something ridiculous: shipping you some hardware.
+
+That hardware is DeskHog, and I can’t wait for you to play with our new developer toy.
+
+## Software, ironically, is the hard bit
+
+DeskHog packages an off-the-shelf, open source board: [Adafruit’s ESP32-S3 Reverse TFT Feather](https://www.adafruit.com/product/5691). Like all of Adafruit’s line, this board is great: it’s got a screen built in, it’s got some buttons, and built-in battery management. 
+
+But man, putting it to use is hard. The internet is packed with code that will demonstrate one feature or another of an ESP32. Actually getting these features working *together* is a whole other problem.
+
+Compounding this issue, the ESP32-S3 is much more resource-constrained than most developers are used to. You have to plan carefully, or risk crashing the board altogether.
+
+DeskHog solves this with open source firmware. Our code provides:
+
+- Built-in wifi setup
+- A web portal for configuration
+- A custom UI built on paginated cards, with button handling
+- Graphics via the *vast* LVGL project
+- Networking examples
+- Stability despite all these tasks running simultaneously
+
+## Your bug is your bug
+
+That last detail is essential. There’s so much going on in microcontroller land. It would be easy to drown as a newbie.
+
+That was me a couple years back.
+
+So we’ve done the homework. An ESP32-S3 has two cores: tasks are divided between them, with networking on one, and the user-facing application running on the other. It’s easy to trip yourself on the classic horrors of concurrency this way and crash the board. DeskHog’s solution: an event queue that can safely pass messages between tasks without mangling memory.
+
+I’ve got a DeskHog sitting on my desk running for days on end without drama.
+
+Which means if you write some new code for DeskHog, you know any crash that shows up is a result of the new code, not some obscure detail lurking in a system you barely understand.
+
+## Built for developers – of all experience levels
+
+You know what sticks in my craw about example microcontroller code?
+
+At its worst, it’s just thousands of lines of code in one file. I cannot work like that. I need tidy encapsulation, separation of concerns – and because it *is C++* – an object-oriented approach to building new features.
+
+If you’re like me, you’re going to feel right at home in the DeskHog project. You’ll find folders to organize network code, UI code, network clients and parsers… it’s all laid out the way you’d expect from building a modern application.
+
+But maybe you’re not like me. Maybe you don’t care about any of that, and you just think hardware is neat. We’ve got you covered there, too.
```

Comment:
```suggestion
And if you're not an experienced dev? If you don't care about tiny encapsulation and an object-orientated approach? We’ve got you covered there, too.
```

Reply:
I think the key change I'm trying to make here is flipping the script to focus on the **user** and how the features benefit them, rather than leading with what bothers you about the straw-man example. The key message for this section should be: DeskHog works at both skill levels and that's a feature, not that you've got an issue with example code.